crates/storage/src/cache.rs

```diff
@@ -20,17 +20,17 @@ pub enum SlotState<T> {
     Ready(T),
 
     /// A database fetch is happening in the background and it will be updated.
-    Pending(broadcast::Receiver<T>),
+    Pending(broadcast::Receiver<Option<T>>),
```

Comment:
I don't get it. Aren't we replacing the `Option<T>` in this PR?
Or are we caching a `T` and maybe we don't have it cached and we return `None`?

Reply:
Oh, makes sense.